tests/test_uri_converters.py

```diff
@@ -64,6 +64,40 @@ def test_int_converter_invalid_config(num_digits):
         converters.IntConverter(num_digits)
 
 
+@pytest.mark.parametrize(
+    'value, min, max, expected',
+    [
+        ('123', None, None, 123.0),
+        ('01', None, None, 1.0),
+        ('001', None, None, 1.0),
+        ('0', None, None, 0.0),
+        ('00', None, None, 00.0),
+        ('1', None, None, 1.0),
+        ('12', 2, None, 12.0),
+        ('1', 1, 1, 1.0),
+        ('12', 1, 20, 12.0),
+        ('12', 2, 10, None),
+        ('1', 2, 13, None),
```

Comment:
I think we could add some additional cases:
- Negative values
- float values (we are only passing ints)
- inf, -inf and nan